### ISSUE TITLE
discourse expressions with subjects

### ISSUE THREAD

**Issue by nschneid** [UniversalDependencies/UD_English-EWT#565]
The subject child rule in https://github.com/amir-zeldes/gum/commit/efab974 flags "I guess", "I mean", and "God forbid" attaching as `discourse`. See https://github.com/UniversalDependencies/docs/issues/1066#issuecomment-2466531594. Should the rule be relaxed for these?

In

- You might be able to get a good deal on some nice furniture -- which I did -- but they're not very communicative and **god forbid** something goes wrong, you'll have to fight to get it resolved. 

it almost seems like "god forbid" takes the place of "if", so should it be a ~fixed expression~ functioning as `mark`?? (UPDATE: actually maybe just the verb attaching as `mark` instead of `discourse`, cf. "provided")

**Comment by amir-zeldes** [comment]
> it almost seems like "god forbid" takes the place of "if",

I think treating it as a `mark` is too avant-garde for my taste. TBH I think this would still be compatible with using "if" alongside, so I wouldn't say it has the same category as "if". And can't you also post-pose the God forbid in the same sentence? Mark shouldn't be able to do that. Plus there are plenty of conditionals like that without "if" and without "God forbid". I would treat this as discourse and an asyndetic advcl, we already have some of those.

**Comment by nschneid** [comment]
> I think this would still be compatible with using "if" alongside, so I wouldn't say it has the same category as "if". And can't you also post-pose the God forbid in the same sentence?

I don't think you can simply omit "god forbid" here. In conversation maybe you could imply an "if" at the beginning of an utterance, but it looks off to do so in the middle of the sentence.

Moving it to the end—"but they're not very communicative and something goes wrong god forbid, you'll have to fight to get it resolved."—looks as weird as simply dropping it.

Yes the more basic function of "god forbid" is as an interjection-like discourse marker, so that holds if "if" is present.

> I would treat this as discourse and an asyndetic advcl, we already have some of those.

Do you have a good example with a finite verb attaching as advcl? https://universal.grew.fr/?custom=67796187c8c36 is mostly turning up errors and constructions like "so X that Y".

**Comment by amir-zeldes** [comment]
> I don't think you can simply omit "god forbid" here. 

I'm pretty sure you can, you just need a nice phatic introducer to get the sentence started. How about "You know, something goes wrong, you'll have to fight to get it resolved"? It's similar to these real examples:

  * Proposal happens in an Escape Room, there's no longer a game (about an escape room, where the manager doesn't like people using it for proposals)
  * A little bit of frost, it's gone. (about basil being sensitive)

I could easily add "God forbid" to either of those:

  * God forbid a proposal happens in an escape room, there's no longer a game
  * God forbid a little bit of frost, it's gone
 
> Moving it to the end looks as weird as simply dropping it.

It's maybe not the preferred variant, but you can find cases like that. It's easier if the protasis is short, but I think that makes sense based on general short-before-long preference. How about this?

  * a little bit of frost God forbid, it's gone

It's also not hard to find real examples of post-posed God forbid, for example:

  * Maybe the election has already been stolen , God forbid . (encow)

> Do you have a good example with a finite verb attaching as advcl

Sure, the one I quoted above is like that (it's from Reddit so wouldn't be included in your link):

![image](https://github.com/user-attachments/assets/c09911d5-be40-4aa2-a6b0-c228354539e6)

**Comment by nschneid** [comment]
Those attestations fit what I meant by:

> In conversation maybe you could imply an "if" at the beginning of an utterance, but it looks off to do so in the middle of the sentence.

If somebody wrote "a little bit of frost God forbid, it's gone" I would simply be confused. If somebody spoke it with the right intonation, stressing FROST, it would make sense with "God forbid" as an optional discourse marker.

I'm not arguing that "God forbid" is always or even usually a subordinator; I just have the sense that it licenses otherwise bare conditional clauses in environments where they would otherwise require another explicit marker like "if".

But let's suppose this is too subtle to worry about for annotation. Should "forbid" attach as `discourse` and have a subject? Our English validation rule now complains about that.

**Comment by amir-zeldes** [comment]
> Should "forbid" attach as discourse and have a subject? Our English validation rule now complains about that.

Yeah, that sounds reasonable to me!

**Comment by nschneid** [comment]
@amir-zeldes updated neaten.py. May be worth updating the GUM one for consistency.

**Comment by amir-zeldes** [comment]
Added locally, will propagate for the next release
